filters/shedder/admission.go

```diff
@@ -0,0 +1,366 @@
+package shedder
+
+import (
+	"context"
+	"math"
+	"math/rand"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/metrics"
+	"go.uber.org/atomic"
+)
+
+func getIntArg(a interface{}) (int, error) {
+	if i, ok := a.(int); ok {
+		return i, nil
+	}
+
+	if f, ok := a.(float64); ok {
+		return int(f), nil
+	}
+
+	return 0, filters.ErrInvalidFilterParameters
+}
+
+func getDurationArg(a interface{}) (time.Duration, error) {
+	if s, ok := a.(string); ok {
+		return time.ParseDuration(s)
+	}
+
+	i, err := getIntArg(a)
+	return time.Duration(i) * time.Millisecond, err
+}
+
+func getFloat64Arg(a interface{}) (float64, error) {
+	if f, ok := a.(float64); ok {
+		return f, nil
+	}
+
+	return 0, filters.ErrInvalidFilterParameters
+}
+
+func getModeArg(a interface{}) (mode, error) {
+	s, ok := a.(string)
+	if !ok {
+		return 0, filters.ErrInvalidFilterParameters
+	}
+	switch s {
+	case "active":
+		return active, nil
+	case "inactive":
+		return inactive, nil
+	case "log":
+		return log, nil
+	}
+
+	return 0, filters.ErrInvalidFilterParameters
+}
+
+type mode int
+
+const (
+	inactive mode = iota
```

Comment:
Maybe `iota + 1` is better because then 0 (that may pop up by mistake somewhere as default int) is not a invalid mode

Reply:
This was a chosen default to make it inactive by default